hudi-utilities/src/test/java/org/apache/hudi/utilities/sources/helpers/TestCheckpointUtils.java

```diff
@@ -104,26 +105,35 @@ public void testComputeOffsetRangesWithoutMinPartitions() {
     ranges = CheckpointUtils.computeOffsetRanges(makeOffsetMap(new int[] {0, 1}, new long[] {200000, 250000}),
         makeOffsetMap(new int[] {0, 1, 2}, new long[] {200010, 350000, 10000}), 100000, 0);
     assertEquals(100000, CheckpointUtils.totalNewMessages(ranges));
+    assertEquals(5, ranges.length);
+    assertEquals(0, ranges[0].partition());
     assertEquals(10, ranges[0].count());
-    assertEquals(89990, ranges[1].count());
-    assertEquals(10000, ranges[2].count());
```

Comment:
The skew caused by the current approach the different can be more if the last partition to be exhausted has the highest number of backlog messages to be consumed.